src/components/SquareBox/SquareBox.tsx

```diff
@@ -37,33 +37,44 @@ interface Shape {
 
 export function SquareBox() {
   const [squares, setSquares] = React.useState<Array<Shape>>([]);
+  const deleteSquare = () => {
+    squares.pop();
+    setSquares(() => [...squares]);
+  };
+  const checkArray = () => {
+    // console.log(squares.filter((item) => item.id !== squares[squares.length - 1].id));
+    setSquares(() => [...squares.filter((item) => item.id !== squares[squares.length - 1].id)]);
+  };
   const addCircle = () => {
     setSquares((prevValue) => ([
       ...prevValue, { color: randomElement(Object.values(COLORS)), shape: 'circle', id: nanoid() },
     ]
     ));
-  };
-
-  const deleteSquare = () => {
-    squares.pop();
-    setSquares(() => [...squares]);
+   // console.log(squares[squares.length - 1].id); // whay don't we have the newest circle in array?
+     setTimeout(() => { checkArray(); }, 1000);
+console.log(squares);
+   // const deletShape= (id) => {squares.filter()}
   };
   const addSquare = React.useCallback(
     () => {
       setSquares((prevValue) => ([
         ...prevValue,
         { color: randomElement(Object.values(COLORS)), shape: 'square', id: nanoid() }]
       ));
+        // setTimeout(() => { setSquares(() => [...squares]); }, 1000);
+       console.log(squares);// empty array
+       setTimeout(() => { checkArray(); }, 1000);
```

Comment:
The idea is good, but you have to structure it better.

First create another function deleteShapeById(id: string), then use this function here with id that is generated above.

In mentioned function filter all shapes that id is different than passed as argument one.